packages/calypso-e2e/src/lib/flows/site-assembler-flow.ts

```diff
@@ -26,35 +22,20 @@ export class SiteAssemblerFlow {
 	 * @param {string} text User-visible text on the button.
 	 */
 	async clickButton( text: string ): Promise< void > {
-		await this.page.click( selectors.button( text ) );
-	}
-
-	/**
-	 * Select Header
-	 */
-	async selectHeader(): Promise< void > {
-		await this.page.getByText( 'Header' ).click();
-		const header = this.page.locator( selectors.blockRenderer ).nth( 0 );
-		await header.click();
+		await this.page.getByRole( 'button', { name: text } ).click();
 	}
 
 	/**
-	 * Select Footer
+	 * Given two parameters, type and index, selects a layout component matching
+	 * the specifications.
+	 *
+	 * @param {LayoutType} type Type of the layout component.
+	 * @param {number} index Index of the item to choose. Defaults to 0.
 	 */
-	async selectFooter(): Promise< void > {
-		await this.page.getByText( 'Footer' ).click();
-		const footer = this.page.locator( selectors.blockRenderer ).nth( 0 );
-		await footer.click();
-	}
+	async selectLayoutComponent( type: LayoutType, index = 0 ): Promise< void > {
+		await this.page.getByRole( 'button', { name: type } ).click();
+		await this.page.waitForLoadState( 'networkidle' );
 
-	/**
-	 * Click "Continue" and land on the Site Editor
-	 */
-	async gotoSiteEditor(): Promise< void > {
-		// Wait for the "Continue" button to be enabled.
-		// @see: https://github.com/Automattic/wp-calypso/pull/75606
-		const waitFor = ( ms: number ) => new Promise( ( resolve ) => setTimeout( resolve, ms ) );
-		await waitFor( 500 );
```

Comment:
https://github.com/Automattic/wp-calypso/pull/75807 made this change; it is to wait for the "Continue" button to be enabled, as its description says.
In this PR, we seem to be using `await page.waitForLoadState( 'networkidle' );` instead, I'd like to know if it ensures the button gets enabled. 👀

Reply:
@okmttdhr Yes, I'm sorry that my changes seem to undo a lot of what you did in #75807. I should probably explain why first - sorry for that.

Although we don't really have anything written down on this matter, past convention[[1](https://github.com/Automattic/wp-calypso/issues/57513#issuecomment-956540650)] as well as Playwright's own guidance on timeouts [[1](https://playwright.dev/docs/api/class-page#page-wait-for-timeout)] has led to a general avoidance of arbitrary timeouts in framework/POM code. 

That hasn't stopped me from trying to see if an arbitrary timeout _could_ solve problems elsewhere though, so I completely understand why it may look tempting. 

Given @karenroldan's observation that it still seems to fail 2/10 times (an unacceptably high incidence rate), I will take a second look to see if we can work around the weird button load issue in the specs. But if not, it may be required to have the feature team look at the button loading behavior. How does that sound?

Reply:
@worldomonation 
You're right, generally speaking, it's better to avoid using arbitrary timeouts, which cause flaky tests, longer running time, etc, so I have tried the code below, but it did not work well for some reason. So I used "waitFor" (which is not ideal) 🙄 
(In case you missed it, disabling the button in 300ms is done intentionally for UX reasons. https://github.com/Automattic/wp-calypso/pull/75606)

```ts
  const locator = page.locator('#selector');
  await page.waitForFunction(([locator]) => {
    const element = locator.asElement()._element;
    return element && !element.disabled;
  }, locator);
```

> I will take a second look to see if we can work around the weird button load issue in the specs. But if not, it may be required to have the feature team look at the button loading behavior

Sounds good 👍  Thank you for all your cooperation!

Reply:
> disabling the button in 300ms is done intentionally for UX reasons. https://github.com/Automattic/wp-calypso/pull/75606

@okmttdhr Interesting, I was not aware of this reason. It means that we have to check for not just the presence of the `Continue` button but also for it to be active and enabled.

I think I have a possible answer, will iterate on this today.

Reply:
After https://github.com/Automattic/wp-calypso/pull/75865 went live, I was able to remove the `waitForTimeout` and leave just the operation to click on the `Continue` button and after 10 iterations locally, it all seems to work fine!

I'll run another 30 iterations, but I think this should work!